rustler/src/types/binary.rs

```diff
@@ -1,43 +1,105 @@
+//! Safe wrappers around Erlang binaries.
+//!
+//! Rustler provides two types for wrapping Erlang binaries: [`Binary`] and
+//! [`OwnedBinary`].  Both represent a contiguous region `u8`s, and they both use
+//! the Erlang allocator. The primary difference between the two is ownership
+//! semantics.
+//!
+//! The _owned_ in `OwnedBinary` refers to the fact that it owns the VM binary is
+//! wraps, and the _owner_ of an `OwnedBinary` is free to modify its contents
+//! without fear of access violations or upsetting the VM. Ownership lasts until it
+//! is dropped by either falling out of scope, or converting it into a regular
```

Comment:
Ownership lasts until it is dropped or consumed by converting it into a regular `Binary`...